common/src/main/java/no/difi/meldingsutveksling/config/IntegrasjonspunktProperties.java

```diff
@@ -218,23 +219,31 @@ public boolean isValidVarsling() {
 
     @Data
     public static class Keystore {
+        /**
+         * Type of KeyStore
+         *
+         * Examples: JKS, Windows-MY
+         */
+        private String type = KeyStore.getDefaultType();
 
         /**
          * Keystore alias for key.
          */
         @NotNull
         private String alias;
+
         /**
          * Path of jks file.
+         *
+         * May be empty if type = Windows-MY
          */
-
-        @NotNull
         private Resource path;
+
         /**
          * Password of keystore and entry.
          */
         @NotNull
-        private String password;
+        private String password = "";
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Remove this hard-coded password. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS2068)